File: abc/abc152/abc152_c.cs
// https://atcoder.jp/contests/abc152/submissions/9612044

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var N = int.Parse(Console.ReadLine());
            var P = Console.ReadLine().Split().Select(int.Parse).ToArray();

            var min_value = N + 1;
            var result = 0;
            for (int i = 0; i < N; i++)
            {
                if (P[i] < min_value)
                {
                    result++;
                    min_value = P[i];
                }
                if (min_value == 1)
                {
                    break;
                }
            }
            Console.WriteLine(result);
        }
    }
}

File: abc/abc152/abc152_c.fs
// https://atcoder.jp/contests/abc152/submissions/11925451

let N = stdin.ReadLine() |> int
let P = stdin.ReadLine().Split() |> Array.map int

let rec solver m (r : int) n =
    if n = N + 1 || m = 1 then r
    elif P.[n] < m then solver P.[n] (r + 1) (n + 1)
    else solver m r (n + 1)

solver (N + 1) 0 0 |> stdout.WriteLine
